File: packages/mail-composer/android/src/main/java/io/capawesome/capacitorjs/plugins/mailcomposer/classes/results/ComposeMailResult.java
package io.capawesome.capacitorjs.plugins.mailcomposer.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.mailcomposer.interfaces.Result;

public class ComposeMailResult implements Result {

    @NonNull
    private final String status;

    public ComposeMailResult(@NonNull String status) {
        this.status = status;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("status", status);
        return result;
    }
}

File: packages/mail-composer/ios/Plugin/Classes/Results/ComposeMailResult.swift
import Foundation
import Capacitor

@objc public class ComposeMailResult: NSObject, Result {
    let status: String

    init(status: String) {
        self.status = status
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["status"] = status
        return result as AnyObject
    }
}
